2020-07-03 20:31:02,998 ERROR: Exception on /login [GET] [in d:\udemy free\microblog github\microblog\venv\lib\site-packages\flask\app.py:1892]
Traceback (most recent call last):
  File "d:\udemy free\microblog github\microblog\venv\lib\site-packages\flask\app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "d:\udemy free\microblog github\microblog\venv\lib\site-packages\flask\app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "d:\udemy free\microblog github\microblog\venv\lib\site-packages\flask\app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "d:\udemy free\microblog github\microblog\venv\lib\site-packages\flask\_compat.py", line 39, in reraise
    raise value
  File "d:\udemy free\microblog github\microblog\venv\lib\site-packages\flask\app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "d:\udemy free\microblog github\microblog\venv\lib\site-packages\flask\app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "D:\udemy free\microblog github\microblog\app\routes.py", line 70, in login
    return render_template('login.html', title='Sign In', form=form)
  File "d:\udemy free\microblog github\microblog\venv\lib\site-packages\flask\templating.py", line 140, in render_template
    ctx.app,
  File "d:\udemy free\microblog github\microblog\venv\lib\site-packages\flask\templating.py", line 120, in _render
    rv = template.render(context)
  File "d:\udemy free\microblog github\microblog\venv\lib\site-packages\jinja2\environment.py", line 1090, in render
    self.environment.handle_exception()
  File "d:\udemy free\microblog github\microblog\venv\lib\site-packages\jinja2\environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "d:\udemy free\microblog github\microblog\venv\lib\site-packages\jinja2\_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "D:\udemy free\microblog github\microblog\app\templates\login.html", line 1, in top-level template code
    {% extends "base.html" %}
  File "D:\udemy free\microblog github\microblog\app\templates\base.html", line 32, in top-level template code
    {% block content %}{% endblock %}
  File "D:\udemy free\microblog github\microblog\app\templates\login.html", line 27, in block "content"
    <a href="{{ url_for('reset_password_request') }}">Click to Reset It</a>
  File "d:\udemy free\microblog github\microblog\venv\lib\site-packages\flask\helpers.py", line 370, in url_for
    return appctx.app.handle_url_build_error(error, endpoint, values)
  File "d:\udemy free\microblog github\microblog\venv\lib\site-packages\flask\app.py", line 2216, in handle_url_build_error
    reraise(exc_type, exc_value, tb)
  File "d:\udemy free\microblog github\microblog\venv\lib\site-packages\flask\_compat.py", line 39, in reraise
    raise value
  File "d:\udemy free\microblog github\microblog\venv\lib\site-packages\flask\helpers.py", line 358, in url_for
    endpoint, values, method=method, force_external=external
  File "d:\udemy free\microblog github\microblog\venv\lib\site-packages\werkzeug\routing.py", line 2179, in build
    raise BuildError(endpoint, values, method, self)
werkzeug.routing.BuildError: Could not build url for endpoint 'reset_password_request'. Did you mean 'edit_profile' instead?
2020-07-03 20:42:25,238 INFO: Microblog startup [in D:\udemy free\microblog github\microblog\app\__init__.py:45]
2020-07-03 20:43:35,390 INFO: Microblog startup [in D:\udemy free\microblog github\microblog\app\__init__.py:45]
2020-07-03 20:43:46,192 ERROR: Exception on /reset_password_request [GET] [in d:\udemy free\microblog github\microblog\venv\lib\site-packages\flask\app.py:1892]
Traceback (most recent call last):
  File "d:\udemy free\microblog github\microblog\venv\lib\site-packages\flask\app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "d:\udemy free\microblog github\microblog\venv\lib\site-packages\flask\app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "d:\udemy free\microblog github\microblog\venv\lib\site-packages\flask\app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "d:\udemy free\microblog github\microblog\venv\lib\site-packages\flask\_compat.py", line 39, in reraise
    raise value
  File "d:\udemy free\microblog github\microblog\venv\lib\site-packages\flask\app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "d:\udemy free\microblog github\microblog\venv\lib\site-packages\flask\app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "D:\udemy free\microblog github\microblog\app\routes.py", line 107, in reset_password_request
    title='Reset Password', form=form)
  File "d:\udemy free\microblog github\microblog\venv\lib\site-packages\flask\templating.py", line 138, in render_template
    ctx.app.jinja_env.get_or_select_template(template_name_or_list),
  File "d:\udemy free\microblog github\microblog\venv\lib\site-packages\jinja2\environment.py", line 930, in get_or_select_template
    return self.get_template(template_name_or_list, parent, globals)
  File "d:\udemy free\microblog github\microblog\venv\lib\site-packages\jinja2\environment.py", line 883, in get_template
    return self._load_template(name, self.make_globals(globals))
  File "d:\udemy free\microblog github\microblog\venv\lib\site-packages\jinja2\environment.py", line 857, in _load_template
    template = self.loader.load(self, name, globals)
  File "d:\udemy free\microblog github\microblog\venv\lib\site-packages\jinja2\loaders.py", line 115, in load
    source, filename, uptodate = self.get_source(environment, name)
  File "d:\udemy free\microblog github\microblog\venv\lib\site-packages\flask\templating.py", line 60, in get_source
    return self._get_source_fast(environment, template)
  File "d:\udemy free\microblog github\microblog\venv\lib\site-packages\flask\templating.py", line 89, in _get_source_fast
    raise TemplateNotFound(template)
jinja2.exceptions.TemplateNotFound: reset_password_request.html
2020-07-03 20:45:09,516 INFO: Microblog startup [in D:\udemy free\microblog github\microblog\app\__init__.py:45]
2020-07-03 20:46:20,074 ERROR: Exception on /reset_password_request [POST] [in d:\udemy free\microblog github\microblog\venv\lib\site-packages\flask\app.py:1892]
Traceback (most recent call last):
  File "d:\udemy free\microblog github\microblog\venv\lib\site-packages\flask\app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "d:\udemy free\microblog github\microblog\venv\lib\site-packages\flask\app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "d:\udemy free\microblog github\microblog\venv\lib\site-packages\flask\app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "d:\udemy free\microblog github\microblog\venv\lib\site-packages\flask\_compat.py", line 39, in reraise
    raise value
  File "d:\udemy free\microblog github\microblog\venv\lib\site-packages\flask\app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "d:\udemy free\microblog github\microblog\venv\lib\site-packages\flask\app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "D:\udemy free\microblog github\microblog\app\routes.py", line 103, in reset_password_request
    send_password_reset_email(user)
  File "D:\udemy free\microblog github\microblog\app\email.py", line 25, in send_password_reset_email
    user=user, token=token),
  File "d:\udemy free\microblog github\microblog\venv\lib\site-packages\flask\templating.py", line 138, in render_template
    ctx.app.jinja_env.get_or_select_template(template_name_or_list),
  File "d:\udemy free\microblog github\microblog\venv\lib\site-packages\jinja2\environment.py", line 930, in get_or_select_template
    return self.get_template(template_name_or_list, parent, globals)
  File "d:\udemy free\microblog github\microblog\venv\lib\site-packages\jinja2\environment.py", line 883, in get_template
    return self._load_template(name, self.make_globals(globals))
  File "d:\udemy free\microblog github\microblog\venv\lib\site-packages\jinja2\environment.py", line 857, in _load_template
    template = self.loader.load(self, name, globals)
  File "d:\udemy free\microblog github\microblog\venv\lib\site-packages\jinja2\loaders.py", line 115, in load
    source, filename, uptodate = self.get_source(environment, name)
  File "d:\udemy free\microblog github\microblog\venv\lib\site-packages\flask\templating.py", line 60, in get_source
    return self._get_source_fast(environment, template)
  File "d:\udemy free\microblog github\microblog\venv\lib\site-packages\flask\templating.py", line 89, in _get_source_fast
    raise TemplateNotFound(template)
jinja2.exceptions.TemplateNotFound: email/reset_password.txt
2020-07-03 20:47:53,964 INFO: Microblog startup [in D:\udemy free\microblog github\microblog\app\__init__.py:45]
2020-07-03 20:53:48,727 INFO: Microblog startup [in D:\udemy free\microblog github\microblog\app\__init__.py:45]
